File: lib/core/constants/legal_content.dart
/// محتوى قانوني للتطبيق
/// يحتوي على Privacy Policy و Terms of Service
class LegalContent {
  LegalContent._();

  /// سياسة الخصوصية
  static String get privacyPolicy => '''
# Privacy Policy

**Last Updated:** ${DateTime.now().year}-${DateTime.now().month.toString().padLeft(2, '0')}-${DateTime.now().day.toString().padLeft(2, '0')}

## Zero Data Collection

**Sada** is a **decentralized, offline mesh messenger**. We (the developers) **cannot see, store, or process** any of your messages, contacts, or data.

### What We Don't Collect

- **No Messages**: All messages are encrypted end-to-end and stored locally on your device only.
- **No Contacts**: Your contact list exists only on your device.
- **No Metadata**: We don't have servers, so there are no server logs, connection logs, or metadata trails.
- **No Analytics**: We don't track your usage, location, or behavior.
- **No Account Information**: User registration is device-bound and offline-only.

### What You Control

- **Local Storage**: All data (messages, contacts, keys) is stored on your device using **Drift (SQLite)**.
- **Encryption Keys**: Your private keys are stored securely in your device's **Android Keystore** (encrypted storage).
- **Duress Mode**: You can use a "Duress PIN" to access a dummy database, providing plausible deniability.

## Mesh Networking

**Sada** uses **WiFi Direct (P2P)** and **Bluetooth LE** to create a mesh network. Messages are:

1. **Encrypted** using **X25519** key exchange and **XSalsa20-Poly1305** authenticated encryption.
2. **Routed** through other devices in the mesh network to reach the destination.
3. **Not stored** on intermediate devices (they only forward encrypted packets).

### Important Notes

- Messages pass through other users' devices, but they are **encrypted** and cannot be read.
- We cannot control or monitor the mesh network routing.
- Network topology is dynamic and peer-to-peer.

## Open Source

**Sada** is **open source** (MIT License). You can:

- Review the source code: [GitHub Repository](https://github.com/obadadallo95/sada-messenger)
- Audit the encryption implementation.
- Verify that no data collection occurs.

## Your Responsibility

- **Backup Your Keys**: If you lose your device, you lose access to your messages (by design, for security).
- **Secure Your Device**: Use biometric lock and strong PINs.
- **Legal Use**: Do not use Sada for illegal activities.

## Changes to This Policy

We may update this Privacy Policy. Changes will be reflected in the app's "About" screen.

---

**Questions?** Open an issue on [GitHub](https://github.com/obadadallo95/sada-messenger/issues).
''';

  /// شروط الاستخدام
  static String get termsOfService => '''
# Terms of Service

**Last Updated:** ${DateTime.now().year}-${DateTime.now().month.toString().padLeft(2, '0')}-${DateTime.now().day.toString().padLeft(2, '0')}

## Alpha Software Disclaimer

**Sada** is currently in **Alpha** (v1.0). This software is provided "as is" without warranty of any kind.

### Use at Your Own Risk

- **No Guarantees**: We do not guarantee message delivery, network availability, or data persistence.
- **No Support**: This is experimental software. Use at your own risk.
- **No Liability**: The developers are not liable for any loss of data, messages, or damages.

## Acceptable Use

### You May:

- Use Sada for personal, non-commercial communication.
- Review and modify the source code (MIT License).
- Share the app with others (offline APK sharing).

### You Must Not:

- **Illegal Activities**: Do not use Sada for illegal purposes, including but not limited to:
  - Terrorism or violence
  - Drug trafficking
  - Human trafficking
  - Fraud or scams
  - Harassment or threats

- **Harmful Content**: Do not share:
  - Malware or viruses
  - Child exploitation material
  - Hate speech or discrimination

- **Network Abuse**: Do not:
  - Intentionally disrupt the mesh network
  - Spam or flood the network
  - Attempt to decrypt others' messages

## Open Source License

**Sada** is licensed under the **MIT License**:

```
MIT License

Copyright (c) 2025 Obada Dallo

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
```

## No Warranty

**THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.**

## Termination

We reserve the right to terminate or restrict access to the software for users who violate these terms.

## Changes to Terms

We may update these Terms of Service. Continued use of the app constitutes acceptance of the updated terms.

---

**Questions?** Open an issue on [GitHub](https://github.com/obadadallo95/sada-messenger/issues).
''';
}

File: app/src/main/kotlin/org/sada/messenger/core/constants/LegalContent.kt
package org.sada.messenger.core.constants

object LegalContent {
    fun about(isArabic: Boolean): String {
        return if (isArabic) {
            """
صدى (Sada) هو تطبيق مراسلة شبكي لا مركزي يعمل بدون إنترنت.

الفكرة الأساسية:
- كل هاتف يعمل كعقدة في الشبكة.
- الرسائل تنتقل بنمط Store-Carry-Forward.
- التشفير من طرف إلى طرف يمنع قراءة المحتوى من أي وسيط.

ما الذي يميز صدى:
- يعمل في البيئات الضعيفة أو المنقطعة.
- لا يعتمد على خوادم مركزية للرسائل.
- يدعم الخصوصية أولاً مع مفاتيح محلية وتخزين محلي.

الإصدار الحالي:
- Android Native Build
- مصمم للاتصالات الميدانية والمرونة التشغيلية.
            """.trimIndent()
        } else {
            """
Sada is a decentralized mesh messenger designed to work without the internet.

Core model:
- Each phone acts as a network node.
- Messages follow Store-Carry-Forward routing.
- End-to-end encryption protects message content from intermediaries.

Why Sada:
- Works in low-connectivity or blackout conditions.
- No dependency on central messaging servers.
- Privacy-first architecture with local keys and local storage.

Current build:
- Android Native Build
- Built for field resilience and operational continuity.
            """.trimIndent()
        }
    }

    fun privacyPolicy(isArabic: Boolean): String {
        return if (isArabic) {
            """
سياسة الخصوصية

آخر تحديث: 2026-02-26

1) جمع البيانات
- لا نجمع رسائلك أو جهات الاتصال أو محتوى محادثاتك.
- لا توجد خوادم مركزية لتخزين الميتاداتا الخاصة بالمراسلة.

2) التشفير
- الرسائل تُشفّر من طرف إلى طرف.
- الأجهزة الوسيطة تنقل الحزم المشفرة فقط ولا تستطيع قراءتها.

3) التخزين
- البيانات الحساسة محفوظة محلياً على جهازك.
- مفاتيح التشفير تبقى على جهازك.

4) التحكم
- يمكنك تفعيل قفل التطبيق ورمز PIN.
- يمكنك حذف البيانات المحلية من جهازك.

5) الشفافية
- صدى مبني بفلسفة صفر-ثقة مع هندسة لا مركزية.
            """.trimIndent()
        } else {
            """
Privacy Policy

Last Updated: 2026-02-26

1) Data collection
- We do not collect your messages, contacts, or chat content.
- No central messaging servers are used to store mesh metadata.

2) Encryption
- Messages are end-to-end encrypted.
- Intermediate devices only carry encrypted packets and cannot read them.

3) Storage
- Sensitive data is stored locally on your device.
- Encryption keys remain on your device.

4) User control
- You can enable app lock and PIN protection.
- You can remove local app data from your device.

5) Transparency
- Sada follows a zero-trust, decentralized architecture.
            """.trimIndent()
        }
    }

    fun intellectualProperty(isArabic: Boolean): String {
        return if (isArabic) {
            """
الترخيص وحقوق الملكية الفكرية

رخصة GNU العمومية الإصدار 3 (GPL v3)
حقوق النشر © 2026 — عبادة دللو (Obada Dallo)

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
1) المشروع مفتوح المصدر
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

تطبيق "صدى" (Sada) هو مشروع مفتوح المصدر مُرخّص بموجب رخصة GNU العمومية الإصدار 3 (GPL v3).

المطور الأصلي والمؤسس: عبادة دللو (Obada Dallo)

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
2) ما يحق لك فعله
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✅ استخدام التطبيق لأي غرض (شخصي أو تجاري).
✅ نسخ وتوزيع الكود المصدري.
✅ تعديل الكود وإنشاء مشاريع مشتقة (Fork).
✅ دمج أجزاء من الكود في مشاريع أخرى.
✅ دراسة الكود والتعلم منه.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
3) الشروط الملزمة
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

⚠️ يجب الالتزام بالشروط التالية:

• ذكر المصدر الأصلي — يجب الإشارة إلى أن العمل الأصلي من تطوير عبادة دللو (Obada Dallo) مع رابط المستودع الأصلي.

• النشر بنفس الرخصة — أي مشروع مشتق أو معدّل يجب أن يكون مفتوح المصدر تحت GPL v3 أيضاً.

• مشاركة الكود المصدري — إذا وزّعت نسخة معدّلة، يجب أن توفّر الكود المصدري الكامل لها.

• الإبقاء على إشعارات حقوق النشر — لا يجوز إزالة أو تعديل إشعارات حقوق النشر الأصلية.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
4) إخلاء المسؤولية
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

- التطبيق مقدم "كما هو" دون أي ضمان صريح أو ضمني.
- المطور غير مسؤول عن أي ضرر مباشر أو غير مباشر ناتج عن الاستخدام.
- المستخدم يتحمل كامل المسؤولية عن استخدامه.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
5) لماذا GPL v3؟
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

صدى مشروع أمن وخصوصية — نؤمن أن الشفافية تبني الثقة. رخصة GPL v3 تضمن أن أي نسخة معدّلة تبقى مفتوحة ليتمكن الجميع من التحقق من أمانها.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
6) التواصل
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

• GitHub Discussions: github.com/obadadallo95/sada-messenger/discussions
• Telegram: @obada_dallo95
• LinkedIn: Obada Dallo

            """.trimIndent()
        } else {
            """
License & Intellectual Property

رخصة GNU العمومية الإصدار 3 (GPL v3)
حقوق النشر © 2026 — عبادة دللو (Obada Dallo)

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
1) المشروع مفتوح المصدر
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

تطبيق "صدى" (Sada) هو مشروع مفتوح المصدر مُرخّص بموجب رخصة GNU العمومية الإصدار 3 (GPL v3).

المطور الأصلي والمؤسس: عبادة دللو (Obada Dallo)

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
2) ما يحق لك فعله
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✅ استخدام التطبيق لأي غرض (شخصي أو تجاري).
✅ نسخ وتوزيع الكود المصدري.
✅ تعديل الكود وإنشاء مشاريع مشتقة (Fork).
✅ دمج أجزاء من الكود في مشاريع أخرى.
✅ دراسة الكود والتعلم منه.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
3) الشروط الملزمة
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

⚠️ يجب الالتزام بالشروط التالية:

• ذكر المصدر الأصلي — يجب الإشارة إلى أن العمل الأصلي من تطوير عبادة دللو (Obada Dallo) مع رابط المستودع الأصلي.

• النشر بنفس الرخصة — أي مشروع مشتق أو معدّل يجب أن يكون مفتوح المصدر تحت GPL v3 أيضاً.

• مشاركة الكود المصدري — إذا وزّعت نسخة معدّلة، يجب أن توفّر الكود المصدري الكامل لها.

• الإبقاء على إشعارات حقوق النشر — لا يجوز إزالة أو تعديل إشعارات حقوق النشر الأصلية.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
4) إخلاء المسؤولية
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

- التطبيق مقدم "كما هو" دون أي ضمان صريح أو ضمني.
- المطور غير مسؤول عن أي ضرر مباشر أو غير مباشر ناتج عن الاستخدام.
- المستخدم يتحمل كامل المسؤولية عن استخدامه.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
5) لماذا GPL v3؟
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

صدى مشروع أمن وخصوصية — نؤمن أن الشفافية تبني الثقة. رخصة GPL v3 تضمن أن أي نسخة معدّلة تبقى مفتوحة ليتمكن الجميع من التحقق من أمانها.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
6) التواصل
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

• GitHub Discussions: github.com/obadadallo95/sada-messenger/discussions
• Telegram: @obada_dallo95
• LinkedIn: Obada Dallo

            """.trimIndent()
        }
    }

    fun termsOfUse(isArabic: Boolean): String {
        return if (isArabic) {
            """
شروط الاستخدام

آخر تحديث: 2026-02-26

1) طبيعة التطبيق
- صدى برنامج اتصالات لا مركزي وقد يتأثر بجودة البيئة اللاسلكية.

2) الاستخدام المسموح
- للاستخدام القانوني والمسؤول فقط.
- يمنع أي استخدام يسبب ضرراً أو إساءة أو نشاطاً غير قانوني.

3) حدود الضمان
- التطبيق يقدم "كما هو" دون ضمان مطلق لوصول الرسائل.
- المطور غير مسؤول عن أي خسائر ناتجة عن سوء الاستخدام أو ظروف خارجية.

4) الأمن التشغيلي
- المستخدم مسؤول عن حماية جهازه ورموزه السرية.
- المستخدم مسؤول عن إدارة النسخ الاحتياطي المناسبة لبياناته.

5) التعديلات
- قد يتم تحديث الشروط لاحقاً، واستمرار الاستخدام يعني قبول التحديثات.
            """.trimIndent()
        } else {
            """
Terms of Use

Last Updated: 2026-02-26

1) App nature
- Sada is a decentralized communication app and depends on wireless conditions.

2) Allowed use
- Legal and responsible use only.
- Any abusive, harmful, or illegal use is prohibited.

3) Warranty limits
- The app is provided "as is" with no absolute delivery guarantee.
- The developer is not liable for losses caused by misuse or external conditions.

4) Operational security
- Users are responsible for securing devices and secret credentials.
- Users are responsible for their own backup strategy.

5) Changes
- Terms may be updated over time; continued use implies acceptance.
            """.trimIndent()
        }
    }
}
